gap/semigroups/semirms.gi

```diff
@@ -11,6 +12,28 @@
 # this file contains methods for every operation/attribute/property that is
 # specific to Rees 0-matrix semigroups.
 
+InstallImmediateMethod(IsFinite,
+IsReesZeroMatrixSubsemigroup and HasGeneratorsOfSemigroup, 0,
+function(R)
+  local P;
+  P := ReesMatrixSemigroupOfFamily(FamilyObj(R.1));
+  if HasIsFinite(P) and IsFinite(P) then
+    return true;
+  fi;
+  TryNextMethod();
+end);
+
+InstallImmediateMethod(IsFinite,
+IsReesMatrixSubsemigroup and HasGeneratorsOfSemigroup, 0,
+function(R)
+  local P;
+  P := ReesMatrixSemigroupOfFamily(FamilyObj(R.1));
+  if HasIsFinite(P) and IsFinite(P) then
+    return true;
+  fi;
+  TryNextMethod();
+end);
+
 #############################################################################
```

Comment:
Your GAP pull request installs an `ImmediateMethod` for `IsReesMatrixSubsemigroup`, but not for `IsReesZeroMatrixSubsemigroup` (I have both here).

However that might not be a problem - the example which prompted this commit was a Rees matrix subsemigroup, and I only put the 0-matrix code in pro-actively. When I remove this code (and use your GAP changes) the tests still run, so this code is probably unnecessary.

@james-d-mitchell Can you think of a reason why the `ImmediateMethod` would only be necessary for a Rees matrix subsemigroup and not a Rees 0-matrix subsemigroup?